apps/opt.c

```diff
@@ -326,6 +326,30 @@ int opt_int(const char *value, int *result)
     return 1;
 }
 
+static STRSTR_PAIR base[] = {
+    {"0x", "hex"},
+    {"0X", "hex"},
+    {"0", "oct"},
+    {NULL}
+};
+
+#define opt_number_error(v, b) \
+    do { \
+        int i = 0; \
+        while (b[i].prefix != NULL) { \
+            if (strncmp(v, b[i].prefix, strlen(b[i].prefix)) == 0) { \
+                BIO_printf(bio_err, \
+                           "%s: Can't parse \"%s\" as a number in %s\n", \
```

Comment:
(of course, you will have to care about "a" vs "an", but that should be trivial...)